src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -6991,6 +7013,60 @@ static bool dependsOnTypeParameter(TypeParameterSymbol typeParameter1, TypeParam
                 annotation = default;
                 return false;
             }
+
+            NullableFlowState visitNestedTargetTypedConstructs()
+            {
+                switch (conversionOperand)
+                {
+                    case BoundConditionalOperator { WasTargetTyped: true } conditional:
+                        Debug.Assert(_conditionalInfoForConversionOpt is not null && _conditionalInfoForConversionOpt.ContainsKey(conditional));
+                        var info = _conditionalInfoForConversionOpt[conditional];
+                        Debug.Assert(info.Length == 2);
+
+                        var consequence = conditional.Consequence;
+                        (BoundExpression consequenceOperand, Conversion consequenceConversion) = RemoveConversion(consequence, includeExplicitConversions: false);
+                        var consequenceRValue = ConvertConditionalOperandOrSwitchExpressionArmResult(consequence, consequenceOperand, consequenceConversion, targetTypeWithNullability, info[0].ResultType, info[0].State, info[0].EndReachable);
+                        var alternative = conditional.Alternative;
+                        (BoundExpression alternativeOperand, Conversion alternativeConversion) = RemoveConversion(alternative, includeExplicitConversions: false);
+                        var alternativeRValue = ConvertConditionalOperandOrSwitchExpressionArmResult(alternative, alternativeOperand, alternativeConversion, targetTypeWithNullability, info[1].ResultType, info[1].State, info[1].EndReachable);
+
+                        _conditionalInfoForConversionOpt.Remove(conditional);
+
+                        return consequenceRValue.State.Join(alternativeRValue.State);
+
+                    case BoundConvertedSwitchExpression { WasTargetTyped: true } @switch:
+                        Debug.Assert(_conditionalInfoForConversionOpt is not null && _conditionalInfoForConversionOpt.ContainsKey(@switch));
+                        info = _conditionalInfoForConversionOpt[@switch];
```

Comment:
The reuse of a 'var' declared in a previous switch section just feels a little weird to me. Consider wrapping the sections in blocks and giving them their own scope to work with.